python/cudf/cudf/core/dataframe.py

```diff
@@ -4002,9 +4004,7 @@ def scatter_by_map(self, map_index, map_size=None, keep_index=True):
                 "Use an integer array/column for better performance."
             )
 
-        if map_size is None:
-            map_size = map_index.unique_count()
-        else:
+        if "debug" in kwargs and map_size is not None:
```

Comment:
What is the "debug" kwarg here?

Reply:
It is a newly introduced flag which when enabled we want to validate if we have a bad `map_size`.

Reply:
I recommended offline that we only check for a "correct" `map_size` argument if the user explicitly asks to check (`debug` in this case).  However, I am personally fine avoiding the check altogether.

Reply:
Perhaps we can still "check" the correctness of `map_size` by checking that the length of the output is <= map_size?

Reply:
```suggestion
        if kwargs.get("debug", False) == True and map_size is not None:
```